pnkfelix/pagelets/demo-botc-pubnub.html

```diff
@@ -3095,100 +3096,171 @@ <h3 id="removeReminderTitle">Remove Reminder</h3>
                             templateData: { role: roleId },
                             timestamp: Date.now() + sentCount // Slight offset to prevent timestamp collisions
                         };
-                        
+
                         // Display locally first (as "sending")
                         displayMessage(message, true);
                         sentMessages.set(message.timestamp, true);
-                        
+
                         pubnub.publish({
                             channel: currentRoom,
                             message: message
                         });
-                        
+
                         sentCount++;
                     }
                 }
             });
-            
+
             if (sentCount > 0) {
                 alert(`Sent role assignments to ${sentCount} players`);
             } else {
                 alert('No roles to distribute');
             }
         }
 
+        function distributeMinionAndDemonInfo() {
+            let sentCount = 0;
+            let demon = '';
+            let minions = [];
+
+            players.forEach(player => {
+                const roleId = playerRoles[player];
+                if (roleId) {
+                    const role = getRoleById(roleId);
+                    if (role?.team === 'demon') {
+                        demon = player
+                    } else if (role?.team === 'minion') {
+                        minions.push(player)
+                    }
+                }
+            });
+
+            // Send Demon Message
+            const demonMessage = minions.length === 1 ? 'Your minion is ' : 'Your minions are '
+            const message = {
+                gameId: currentGameId,
+                type: 'private_message',
+                messageType: 'info',
+                from: currentUser,
+                to: demon,
+                content: `${demonMessage} ${minions.join(', ')}.`,
+                timestamp: Date.now() + sentCount // Slight offset to prevent timestamp collisions
+            };
+
+            // Display locally first (as "sending")
+            displayMessage(message, true);
+            sentMessages.set(message.timestamp, true);
+
+            pubnub.publish({
+                channel: currentRoom,
+                message: message
+            });
+
+            sentCount++;
+
+            // Send Minion Messages
+            minions.forEach(minion => {
+                const message = {
```

Comment:
You should also send a separate message to the minions saying whom all the minions are, right?